joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala

```diff
@@ -917,9 +917,17 @@ class AstCreator(filename: String, javaParserAst: CompilationUnit, global: Globa
   }
 
   private def expressionReturnTypeFullName(expr: Expression): Option[String] = {
-    tryWithSafeStackOverflow(expr.calculateResolvedType()).toOption
-      .flatMap(typeInfoCalc.fullName)
-      .orElse(exprNameFromStack(expr))
+    val resolvedTypeOption = tryWithSafeStackOverflow(expr.calculateResolvedType()) match {
```

Comment:
Could this not be part of the `flatMap` that already existed?
e.g. 
```scala
tryWithSafeStackOverflow(expr.calculateResolvedType()).flatMap {
     case Failure(ex) =>
         ex match {
           // If ast parser fails to resolve type, try resolving locally by using name
           case symbolException: UnsolvedSymbolException =>
             scopeStack.lookupVariableType(symbolException.getName)
           case _ => None
         }
       case Success(resolvedType) => typeInfoCalc.fullName(resolvedType)
}.orElse(exprNameFromStack(expr))

Reply:
FlatMap when applied on a Try, the statement inside of it only gets executed for Success.
In this case, inside flatMap, I directly get `resolvedType`

Reply:
Ok make sense